auto-self-nominate/index.ts

```diff
@@ -0,0 +1,179 @@
+import { U64 } from "codechain-primitives/lib";
+import { SDK } from "codechain-sdk";
+import { PlatformAddress } from "codechain-sdk/lib/core/classes";
+import { Custom } from "codechain-sdk/lib/core/transaction/Custom";
+
+const RLP = require("rlp");
+
+const ACTION_TAG_SELF_NOMINATE = 4;
+const STAKE_ACTION_HANDLER_ID = 2;
+
+function getConfig(field: string): string {
+    const c = process.env[field];
+    if (c == null) {
+        throw new Error(`${field} is not specified`);
+    }
+    return c;
+}
+
+function rpcUrl(server: string): string {
+    switch (server) {
+        case "corgi":
+            return "https://corgi-rpc.codechain.io/";
+        case "mainnet":
+            return "https://rpc.codechain.io/";
+        default:
+            throw Error("Invalid server configuration");
+    }
+}
+
+function networkId(server: string): string {
+    switch (server) {
+        case "beagle":
+            return "bc";
+        case "corgi":
+            return "wc";
+        case "mainnet":
+            return "cc";
+        default:
+            throw Error("Invalid server configuration");
+    }
+}
+
+const SERVER: string = (() => {
+    const server = process.env.SERVER || "beagle";
+    if (["beagle", "corgi", "mainnet"].indexOf(server) >= 0) {
```

Comment:
It's not flexible.
Please read the threads related to https://github.com/CodeChain-io/cron-jobs/issues/61#issuecomment-501132502

Reply:
Fixed to get `NETWORK_ID` and `RPC_URL` from environment variables.